From 11650d40b2e03240a97c81b2e4b01e88c5272a7e Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Tue, 1 Jun 2010 10:15:06 -0300
Subject: [PATCH 1/5] qxl: drop check for depths != 32.

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1275387307-25378-2-git-send-email-kraxel@redhat.com>
Patchwork-id: 9642
O-Subject: [RHEL-6 kvm PATCH 1/2] qxl: drop check for depths != 32.
Bugzilla: 597198
RH-Acked-by: Alon Levy <alevy@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Juan Quintela <quintela@redhat.com>

Just use 32 bit.  The depth spice uses internally is independant from
the DisplaySurface depth, so this check is bogous in the first place.

bugzilla: #597198 --  qxl: 16bpp vga mode is broken

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 hw/qxl.c |    6 +-----
 1 files changed, 1 insertions(+), 5 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 hw/qxl.c |    6 +-----
 1 files changed, 1 insertions(+), 5 deletions(-)

diff --git a/hw/qxl.c b/hw/qxl.c
index e58158f..b73c087 100644
--- a/hw/qxl.c
+++ b/hw/qxl.c
@@ -249,11 +249,7 @@ static void _qxl_get_info(PCIQXLDevice *d, QXLDevInfo *info)
 
         info->x_res = ds_get_width(qxl_vga.ds);
         info->y_res = ds_get_height(qxl_vga.ds);
-        info->bits = ds_get_bits_per_pixel(qxl_vga.ds);
-        if (info->bits != 32) {
-            dprintf(1, "%s: unexpected depth %d\n", __FUNCTION__, info->bits);
-            abort();
-        }
+        info->bits = 32;
 
         info->use_hardware_cursor = false;
 
-- 
1.7.0.3

